In [1]:
#Import dependancies
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split

In [2]:
# Import the data
path = "Resources/lending_data.csv"
lending_df = pd.read_csv(path)
lending_df.tail()

loan_size  interest_rate  borrower_income  debt_to_income  \
77531    19100.0         11.261            86600        0.653580   
77532    17700.0         10.662            80900        0.629172   
77533    17600.0         10.595            80300        0.626401   
77534    16300.0         10.068            75300        0.601594   
77535    15600.0          9.742            72300        0.585062   

       num_of_accounts  derogatory_marks  total_debt  loan_status  
77531               12                 2       56600            1  
77532               11                 2       50900            1  
77533               11                 2       50300            1  
77534               10                 2       45300            1  
77535                9                 2       42300            1

## Prediction:
I think the Logistic Regression will perform better than the Random Forests Classifier. This is because Random FOrests CLassifier performs better with categorical data than numeric while this data is numeric. Logistic Regression also performs better when there is little noise which I believe is the case in this data.

In [3]:
# Split the data into X_train, X_test, y_train, y_test
X = lending_df.drop("loan_status", axis=1 )
y = lending_df["loan_status"]

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 42)

In [4]:
#Scaling the data provided
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler().fit(X_train)
X_train_scaled = scaler.transform(X_train)
scaler = StandardScaler().fit(X_test)
X_test_scaled = scaler.transform(X_test)

In [5]:
# Train a Logistic Regression model print the model score
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression().fit(X_train_scaled, y_train)
print(f"Training Score: {classifier.score(X_train_scaled, y_train)}")
print(f"Test Score: {classifier.score(X_test_scaled, y_test)}")

Training Score: 0.9941188609162196
Test Score: 0.9941704498555509


In [6]:
# Train a Random Forest Classifier model and print the model score
from sklearn.ensemble import RandomForestClassifier

clf = RandomForestClassifier(random_state = 42, n_estimators = 500)
clf.fit(X_train_scaled, y_train)



print(f"Training Score: {clf.score(X_train_scaled, y_train)}")
print(f"Test Score: {clf.score(X_test_scaled, y_test)}")

Training Score: 0.9971970009629936
Test Score: 0.9929323153115972


## Conclusion
The Random Forest Classifier testing score of 0.99417 is actually higher than the Logistic Regression testing score of 0.9929.
This is contrary to my prediction earlier. 
However, both models appear to be performing well with a high score on both. I think either of them could be adapted for this particular data set given. 
